spec/requests/container_images_spec.rb

```diff
@@ -1,3 +1,16 @@
+class FailedScanContainerImage < ContainerImage
```

Comment:
@moolitayer cc

Reply:
IMO it's not a good idea to stub like this in an integration test. You're testing what happens if a container image raises `Exception` during a scan - which I would hope it would never do. It would be far better to create some state that results in an exception in the actual implementation of `ContainerImage`. If none can be created, it's possible you don't need the exception handling code above.

Reply:
@imtayadeway the `scan` method raises and event to create a Job for the scan. It is not suppose to fail but the rescue is there for completeness. I don't the conditions that will make this fail but they are probably from outside of the `ContainerImage` class. I don't feel strongly either way, do you think the rescue above should be removed?

Reply:
> rescue is there for completeness

It's not complete because you haven't demonstrated why it is needed. I don't feel strongly that you should remove it, but I do feel strongly that you should remove the stubbing here. You are stubbing the SUT here (since it is an integration test), which only serves to demonstrate that your code handles the stub correctly.

Reply:
This is a little similar to [1], This will create an automate event through a series for function calls that will create a scanning Job. I have noticed that no specific exception type is mentioned there and it is not mentioned in any test.

[1] https://github.com/ManageIQ/manageiq-api/blob/master/app/controllers/api/vms_controller.rb#L410